cli/command/service.go

```diff
@@ -10,6 +18,260 @@ func NewServiceCommand(b *CommandBase) *ServiceCommand {
+				Action:    s.logs,
+				ArgsUsage: "SERVICE_NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return (default: 0)",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+			{
+				Name:      "scale",
+				Usage:     "scale a service",
+				Action:    s.scale,
+				ArgsUsage: "SERVICE_NAME COUNT",
+			},
+			{
+				Name:      "update",
+				Usage:     "run a new dploy on a service",
+				Action:    s.update,
+				ArgsUsage: "SERVICE_NAME DEPLOY",
+			},
+		},
+	}
+}
+
+func (s *ServiceCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "SERVICE_NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	deployID, err := s.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := s.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	var loadBalancerID string
+	if loadBalancerName := c.String("loadbalancer"); loadBalancerName != "" {
+		id, err := s.resolveSingleEntityIDHelper("load_balancer", loadBalancerName)
+		if err != nil {
+			return err
+		}
+
+		loadBalancerID = id
+	}
+
+	req := models.CreateServiceRequest{
+		DeployID:       deployID,
+		EnvironmentID:  environmentID,
+		LoadBalancerID: loadBalancerID,
+		ServiceName:    args["SERVICE_NAME"],
+	}
+
+	jobID, err := s.client.CreateService(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFN := func(serviceID string) error {
+		service, err := client.WaitForDeployment(s.client, serviceID, c.GlobalDuration(config.FLAG_TIMEOUT))
+		if err != nil {
+			return err
+		}
+
+		return s.printer.PrintServices(service)
+	}
+
+	return s.waitOnJobHelper(c, jobID, "creating", onCompleteFN)
+}
+
+func (s *ServiceCommand) delete(c *cli.Context) error {
+	deleteFN := func(serviceID string) (string, error) {
+		return s.client.DeleteService(serviceID)
+	}
+
+	return s.deleteHelper(c, "service", deleteFN)
+}
+
+func (s *ServiceCommand) list(c *cli.Context) error {
+	serviceSummaries, err := s.client.ListServices()
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintServiceSummaries(serviceSummaries...)
+}
+
+func (s *ServiceCommand) logs(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "SERVICE_NAME")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["SERVICE_NAME"])
+	if err != nil {
+		return err
+	}
+
+	query := buildLogQueryHelper(serviceID, c.String("start"), c.String("end"), c.Int("tail"))
+
+	logs, err := s.client.ReadServiceLogs(serviceID, query)
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintLogs(logs...)
```

Comment:
@tlake in manual testing, how long have you observed it takes for this command to return?

Reply:
@diemonster I don't recall it taking a particularly long time, but I'll whack at it again and get back to you.

Reply:
```
$ time go run main.go se
rvice logs guestbod0dfc


2017/10/12 19:20:48 Using memory backend
2017/10/12 19:20:48 Listening on :80


real    0m2.781s
user    0m1.312s
sys     0m0.258s
```

So yeah, nothing strange on my test.

Reply:
@diemonster
```
# from layer0/cli/
$ eval $(l0-setup endpoint -d -i tlakedev | sed -n '1!p')
$ export LAYER0_AWS_LOG_GROUP_NAME="l0-tlakedev"
$ export LAYER0_API_ENDPOINT=http://localhost:9090
$ go install .
$ time for i in {1..10}; do go run main.go service logs guestbod0dfc; done
...
real    0m43.647s
user    0m4.905s
sys     0m1.280s
```

So about 4.4 seconds, on average.

Reply:
I've raised this issue on a couple other PRs, it's mostly a UX thing, but if we have commands that tend to take a while to return, we should consider using the spinner functionality to let the user know that something is happening. Unfortunately, we don't have a good pattern for doing this (at the moment, at least) without the corresponding `client` call returning a `job`.